### PR TITLE
feat(privacy.providers): add aws_comprehend as scrubbing provider

### PR DESCRIPTION
**Pull request by KrishPatel13**
**What kind of change does this PR introduce?**
Add AWS Comprehend as a Scrubbing Provider

**Summary**

**Checklist**
* [x] My code follows the style guidelines of OpenAdapt
* [x] I have performed a self-review of my code
* [x] If applicable, I have added tests to prove my fix is functional/effective
* [x] I have linted my code locally prior to submission
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation (e.g. README.md, requirements.txt)
* [x] New and existing unit tests pass locally with my changes

**How can your code be run and tested?**

**Other information**

### CODE REVIEW

**Comment by KrishPatel13** [comment]
Todo fix this error:

```
(openadapt-py3.10) PS P:\OpenAdapt AI - MLDS AI\cloned_repo\my_forked\OpenAdapt> pytest .\tests\openadapt\privacy\providers\test_comprehend_scrub.py
==================================================== test session starts ====================================================
platform win32 -- Python 3.10.11, pytest-7.1.3, pluggy-1.2.0
rootdir: P:\OpenAdapt AI - MLDS AI\cloned_repo\my_forked\OpenAdapt
plugins: anyio-3.7.1
collected 14 items

tests\openadapt\privacy\providers\test_comprehend_scrub.py F.............                                              [100%]

========================================================= FAILURES ==========================================================
_____________________________________________________ test_empty_string _____________________________________________________

    def test_empty_string() -> None:
        """Test empty string input for scrub function returns empty string."""
        text = ""
        expected_output = ""
>       assert scrub.scrub_text(text) == expected_output

tests\openadapt\privacy\providers\test_comprehend_scrub.py:41:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
openadapt\privacy\providers\aws_comprehend.py:99: in scrub_text
    languages = comp_detect.detect_languages(text)
openadapt\privacy\providers\aws_comprehend.py:45: in detect_languages
    response = self.comprehend_client.detect_dominant_language(Text=text)
C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-vs6jmx3S-py3.10\lib\site-packages\botocore\client.py:535: in _api_call
    return self._make_api_call(operation_name, kwargs)
C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-vs6jmx3S-py3.10\lib\site-packages\botocore\client.py:936: in _make_api_call
    request_dict = self._convert_to_request_dict(
C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-vs6jmx3S-py3.10\lib\site-packages\botocore\client.py:1007: in _convert_to_request_dict
    request_dict = self._serializer.serialize_to_request(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <botocore.validate.ParamValidationDecorator object at 0x000001D964F47E80>, parameters = {'Text': ''}
operation_model = OperationModel(name=DetectDominantLanguage)

    def serialize_to_request(self, parameters, operation_model):
        input_shape = operation_model.input_shape
        if input_shape is not None:
            report = self._param_validator.validate(
                parameters, operation_model.input_shape
            )
            if report.has_errors():
>               raise ParamValidationError(report=report.generate_report())
E               botocore.exceptions.ParamValidationError: Parameter validation failed:
E               Invalid length for parameter Text, value: 0, valid min length: 1

C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-vs6jmx3S-py3.10\lib\site-packages\botocore\validate.py:381: ParamValidationError
================================================== short test summary info ================================================== 
FAILED tests/openadapt/privacy/providers/test_comprehend_scrub.py::test_empty_string - botocore.exceptions.ParamValidationE...
=============================================== 1 failed, 13 passed in 8.49s ================================================ 
(openadapt-py3.10) PS P:\OpenAdapt AI - MLDS AI\cloned_repo\my_forked\OpenAdapt> 
```

**Comment by KrishPatel13** [comment]
After Visualization:
image

**Comment by KrishPatel13** [comment]
```
(openadapt-py3.10) PS P:\OpenAdapt AI - MLDS AI\cloned_repo\my_forked\OpenAdapt> pytest
========================================================================================= test session starts =========================================================================================
platform win32 -- Python 3.10.11, pytest-7.1.3, pluggy-1.2.0
rootdir: P:\OpenAdapt AI - MLDS AI\cloned_repo\my_forked\OpenAdapt
plugins: anyio-3.7.1
collected 43 items

tests\openadapt\test_crop.py .                                                                                                                                                                   [  2%]
tests\openadapt\test_events.py .......                                                                                                                                                           [ 18%]
tests\openadapt\test_summary.py ..                                                                                                                                                               [ 23%]
tests\openadapt\privacy\test_providers.py .                                                                                                                                                      [ 25%]
tests\openadapt\privacy\providers\test_comprehend_scrub.py ...............                                                                                                                       [ 60%]
tests\openadapt\privacy\providers\test_presidio_scrub.py .................                                                                                                                       [100%]

========================================================================================== warnings summary ===========================================================================================
C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-vs6jmx3S-py3.10\lib\site-packages\pycountry\__init__.py:10
  C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-vs6jmx3S-py3.10\lib\site-packages\pycountry\__init__.py:10: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    import pkg_resources

C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-vs6jmx3S-py3.10\lib\site-packages\pkg_resources\__init__.py:2871
  C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-vs6jmx3S-py3.10\lib\site-packages\pkg_resources\__init__.py:2871: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('mpl_toolkits')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(pkg)

C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-vs6jmx3S-py3.10\lib\site-packages\pkg_resources\__init__.py:2871
  C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-vs6jmx3S-py3.10\lib\site-packages\pkg_resources\__init__.py:2871: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('sphinxcontrib')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(pkg)

tests/openadapt/privacy/providers/test_presidio_scrub.py::test_scrub_image
  C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-vs6jmx3S-py3.10\lib\site-packages\pytesseract\pytesseract.py:382: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    return LooseVersion(

tests/openadapt/privacy/providers/test_presidio_scrub.py::test_scrub_image
  C:\Users\Krish Patel\AppData\Local\pypoetry\Cache\virtualenvs\openadapt-vs6jmx3S-py3.10\lib\site-packages\setuptools\_distutils\version.py:345: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    other = LooseVersion(other)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=================================================================================== 43 passed, 5 warnings in 56.05s =================================================================================== 
```

**Comment by KrishPatel13** [comment]
@abrichr Ready for Review!

File: openadapt/config.py [hunk 1]
```diff
@@ -91,7 +91,7 @@
     # Calculate and save the difference between 2 neighboring screenshots
     "SAVE_SCREENSHOT_DIFF": False,
     "SPACY_MODEL_NAME": "en_core_web_trf",
-    "SCRUB_PROVIDER_NAME": ["Presidio"],
+    "SCRUB_PROVIDER_NAME": ["presidio", "comprehend"],
```
Comment (KrishPatel13) [inline openadapt/config.py:94]: add comprehend to the provider list

Comment (abrichr) [reply openadapt/config.py:94]: Can you please implement these as properties on an enum-style class? e.g.

```
class ScrubProvider:
    PRESDIO = "PRESIDIO"
    COMPREHEND = "COMPREHEND"
```

Note you can always do something like `ScrubProvider.COMPREHEND.lower()`

File: openadapt/privacy/providers/aws_comprehend.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+# snippet-end:[python.example_code.comprehend.DetectPiiEntities]
+
+
+class ComprehendScrubbingProvider(
+    ScrubbingProvider, TextScrubbingMixin
+):  # pylint: disable=abstract-method
+    """A Class for AWS Comprehend Scrubbing Provider."""
+
+    name: str = config.SCRUB_PROVIDER_NAME[1]  # pylint: disable=E1101
```
Comment (abrichr) [inline openadapt/privacy/providers/aws_comprehend.py:74]: What do you think about changing this to something like:

```
[nam](name: str = config.SCRUB_PROVIDER.COMPREHEND ...
```

**Comment by KrishPatel13** [comment]
Next-up:

- [ ] Edit AWS Comprehend class's config to add enum classes names as pointed out here: https://github.com/OpenAdaptAI/OpenAdapt/pull/476#discussion_r1304821459
- [ ] Add Private AI Scrubbing class for Text, Image, PDF, and MP4 if it supports

File: openadapt/privacy/providers/aws_comprehend.py (hunk 2 above)
Comment (KrishPatel13) [reply openadapt/privacy/providers/aws_comprehend.py]: Addressed in: [09750f3](https://github.com/OpenAdaptAI/OpenAdapt/pull/476/commits/09750f33cd49619c5dd2860065d892cac34aa0bd)

File: openadapt/config.py (hunk 1 above)
Comment (KrishPatel13) [reply openadapt/config.py]: [09750f3](https://github.com/OpenAdaptAI/OpenAdapt/pull/476/commits/09750f33cd49619c5dd2860065d892cac34aa0bd)

Comment (KrishPatel13) [reply openadapt/config.py]: Addressed above

**Comment by KrishPatel13** [comment]
Ready for merging!

**Review by 0dm — APPROVED** [review]

**Pull request merged**